google/generativeai/retriever.py

```diff
@@ -30,14 +30,11 @@
 from google.generativeai.types import safety_types
 from google.generativeai.types.model_types import idecode_time
 
-_CORPORA_NAME_REGEX = re.compile(r"^corpora/[a-z0-9-]+")
-_REMOVE = string.punctuation
-_REMOVE = _REMOVE.replace("-", "")  # Don't remove hyphens
-_PATTERN = r"[{}]".format(_REMOVE)  # Create the pattern
+_VALID_NAME = r"^[^-][a-zA-Z\d-][^-]+$"
```

Comment:
This regex doesn't seem right.

Start with one "anything except a `-`" character
Followed by "One(!) alphanumeric character, or '-'"
Followed by "at least one non '-' character"

Reply:
The API spec also says "lowercase alphanumeric"

Reply:
maybe copy the regex from the proto definition?